hydromt/components/geoms.py

```diff
@@ -0,0 +1,171 @@
+"""Table component."""
+
+import os
+from glob import glob
+from os.path import dirname, isdir, join
+from pathlib import Path
+from typing import (
+    TYPE_CHECKING,
+    Dict,
+    Optional,
+    Union,
+    cast,
+)
+
+import geopandas as gpd
+from geopandas import GeoDataFrame, GeoSeries
+
+from hydromt.components.base import ModelComponent
+from hydromt.hydromt_step import hydromt_step
+from hydromt.metadata_resolver import ConventionResolver
+
+if TYPE_CHECKING:
+    from hydromt.models.model import Model
+
+_DEFAULT_GEOMS_FILENAME = "geoms/{name}.geojson"
+
+
+class GeomsComponent(ModelComponent):
+    """A component to manage geo-spatial geometries."""
```

Comment:
I wonder which docstrings is the best place to describe the component. Is it the class or the __init__ one?
I would be nice if at least mentions the data property and what it returns (dict of geopandas.GeoDataFrame).

Ideally we would have something like [geopandas](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.html) or [xarray](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.html) too at some point.

Reply:
I think the class docstring is better since that will be what is provided on a hover in the code, but we can certainly expand or debate this definition at some point. I've added a mention of the dict of geodataframes for now at least.